kernel/src/hil/hw_debug.rs

```diff
@@ -0,0 +1,31 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
```

Comment:
Having (eventually) one `hw_debug` file with a bunch of HILs feels like a weird organization to me. Seems like this should really nest under `hil/debug/hw_counter` or something like that; but maybe that's a nuance for the day we have >1 debug-related HIL.

Reply:
This HIL is sort of intended to correspond to anything you would implement on top of Arm's debug control block, which groups various hardware debug functionality. So having a few traits in a single file kinda makes sense because they should be closely related. But I agree that is something we can revisit if we find it does not map well to RISC-V, etc.